components/core/tests/test-hash_utils.cpp

```diff
@@ -0,0 +1,50 @@
+#include <string_view>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp/ErrorCode.hpp"
+#include "../src/clp/hash_utils.hpp"
+#include "../src/clp/type_utils.hpp"
+using clp::convert_hash_to_hex_string;
+using clp::ErrorCode_Success;
+using clp::get_hmac_sha256_hash;
+using clp::get_sha256_hash;
+using clp::size_checked_pointer_cast;
+using std::string_view;
+using std::vector;
+
+TEST_CASE("test_sha256", "[hash_utils]") {
+    constexpr string_view input_string{"ThisIsARandomTestInput"};
+    constexpr string_view reference_sha256{
+            "c3a1d9f04ada1198c4c63bf51d9933fc2cc216429275cadabdcb2178775add0c"
+    };
+    vector<unsigned char> hash{};
+
+    CHECK(ErrorCode_Success
```

Comment:
For `CHECK`, execution continues in the same test case even if the assertion fails.
In this case, if the hash generation fails there's no reason to check the result. I think we should use `REQUIRE`